include/message_filters/subscriber.h

```diff
@@ -68,6 +70,47 @@ class SubscriberBase
    * \param qos (optional) The rmw qos profile to use to subscribe
    */
   virtual void subscribe(rclcpp::Node * node, const std::string& topic, const rmw_qos_profile_t qos = rmw_qos_profile_default) = 0;
+
+  /**
+   * \brief Subscribe to a topic.
+   *
+   * If this Subscriber is already subscribed to a topic, this function will first unsubscribe.
+   * This override allows SubscriptionOptions to be passed into the class without changing API.
+   *
+   * \param node The rclcpp::Node::SharedPtr to use to subscribe.
+   * \param topic The topic to subscribe to.
+   * \param qos (optional) The rmw qos profile to use to subscribe.
+   * \param options The subscription options to use to subscribe.
+   */
+  virtual void subscribe(
+    rclcpp::Node::SharedPtr node,
+    const std::string& topic,
+    const rmw_qos_profile_t qos,
```

Comment:
Feel free to ignore

----

I would take advantake and use `rclcpp::QoS` in the new methods.

You can use `qos.get_rmw_qos_profile()` in the default implementation to call the old method.

Reply:
My feeling is to do this in a separate PR.

Reply:
sounds good to me